test/catalog/catalog_test.cpp

```diff
@@ -303,5 +304,105 @@ TEST_F(CatalogTests, DroppingCatalog) {
+  // Drops layout and verifies that the default_layout is reset.
+  // It also queries pg_layout to ensure that the entry is removed.
+
+  auto db_name = "temp_db";
+  auto table_name = "temp_table";
+  auto catalog = catalog::Catalog::GetInstance();
+  // Create database.
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  catalog->CreateDatabase(db_name, txn);
+
+  // Create table.
+  auto val0 = catalog::Column(type::TypeId::INTEGER,
+                              type::Type::GetTypeSize(type::TypeId::INTEGER),
+                              "val0", true);
+  auto val1 = catalog::Column(type::TypeId::INTEGER,
+                              type::Type::GetTypeSize(type::TypeId::INTEGER),
+                              "val1", true);
+  auto val2 = catalog::Column(type::TypeId::INTEGER,
+                              type::Type::GetTypeSize(type::TypeId::INTEGER),
+                              "val2", true);
+  auto val3 = catalog::Column(type::TypeId::INTEGER,
+                              type::Type::GetTypeSize(type::TypeId::INTEGER),
+                              "val3", true);
+  std::unique_ptr<catalog::Schema> table_schema(
+      new catalog::Schema({val0, val1, val2, val3}));
+  catalog->CreateTable(db_name, DEFAULT_SCHEMA_NAME, table_name,
+                       std::move(table_schema), txn);
+  txn_manager.CommitTransaction(txn);
+
+  txn = txn_manager.BeginTransaction();
+  auto table =
+      catalog->GetTableWithName(db_name, DEFAULT_SCHEMA_NAME, table_name, txn);
+  auto table_oid = table->GetOid();
+  auto database_oid = table->GetDatabaseOid();
+  txn_manager.CommitTransaction(txn);
+
+  // Change default layout.
+
+  std::map<oid_t, std::pair<oid_t, oid_t>> default_map;
+  default_map[0] = std::make_pair(0, 0);
+  default_map[1] = std::make_pair(0, 1);
+  default_map[2] = std::make_pair(1, 0);
+  default_map[3] = std::make_pair(1, 1);
+
+  txn = txn_manager.BeginTransaction();
+  auto default_layout =
+      catalog->CreateDefaultLayout(database_oid, table_oid, default_map, txn);
+  txn_manager.CommitTransaction(txn);
+  EXPECT_NE(nullptr, default_layout);
+
+  // Create additional layout.
+  std::map<oid_t, std::pair<oid_t, oid_t>> non_default_map;
+  non_default_map[0] = std::make_pair(0, 0);
+  non_default_map[1] = std::make_pair(0, 1);
+  non_default_map[2] = std::make_pair(1, 0);
+  non_default_map[3] = std::make_pair(1, 1);
+
+  txn = txn_manager.BeginTransaction();
+  auto other_layout =
+      catalog->CreateLayout(database_oid, table_oid, non_default_map, txn);
+  txn_manager.CommitTransaction(txn);
+
+  // Check that the default layout is still the same.
+  EXPECT_NE(*other_layout.get(), table->GetDefaultLayout());
+
+  // Drop the default layout.
+  auto default_layout_oid = default_layout->GetOid();
+  txn = txn_manager.BeginTransaction();
+  EXPECT_EQ(ResultType::SUCCESS, catalog->DropLayout(database_oid, table_oid,
```

Comment:
You said Persist the **HYBRID** layouts to the pg_catalog table. The two pre-defined layouts, **ROW** and **COLUMN** stores are not persisted. 
I may understand this wrong, the word ** Persist** means store this information in catalog table, right? If that is true, why `DropLayout(default_layout_oid)` returns SUCCESS? Since `default_layout` is **ROW** stored, it should not be inserted into pg_layout?

Reply:
Yeah, it makes sense now. Thanks for your explanation